test/integration/apiserver/apiserver_test.go

```diff
@@ -403,6 +406,361 @@ func TestNameInFieldSelector(t *testing.T) {
+			want: func(t *testing.T) {
+				ns := "default"
+				svc, err := clientset.CoreV1().Services(ns).Create(&v1.Service{ObjectMeta: metav1.ObjectMeta{Name: "test-1", Annotations: map[string]string{"foo": "bar"}}, Spec: v1.ServiceSpec{Ports: []v1.ServicePort{{Port: 1000}}}})
+				if err != nil {
+					t.Fatalf("unable to create service: %v", err)
+				}
+
+				cfg := metadata.ConfigFor(&restclient.Config{Host: s.URL})
+				wrapper := &callWrapper{}
+				cfg.Wrap(func(rt http.RoundTripper) http.RoundTripper {
+					wrapper.nested = rt
+					return wrapper
+				})
+
+				client := metadata.NewConfigOrDie(cfg).Resource(v1.SchemeGroupVersion.WithResource("services"))
+				items, err := client.Namespace(ns).List(metav1.ListOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if items.ResourceVersion == "" {
+					t.Fatalf("unexpected items: %#v", items)
+				}
+				for _, item := range items.Items {
+					if item.Name != "test-1" {
+						continue
+					}
+					if item.UID != svc.UID || item.Annotations["foo"] != "bar" {
+						t.Fatalf("unexpected object: %#v", item)
+					}
+				}
+
+				if wrapper.resp == nil || wrapper.resp.Header.Get("Content-Type") != "application/vnd.kubernetes.protobuf" {
+					t.Fatalf("unexpected response: %#v", wrapper.resp)
+				}
+				wrapper.resp = nil
+
+				item, err := client.Namespace(ns).Get("test-1", metav1.GetOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if item.ResourceVersion == "" {
+					t.Fatalf("unexpected items: %#v", items)
+				}
+				if item.UID != svc.UID || item.Annotations["foo"] != "bar" {
+					t.Fatalf("unexpected object: %#v", item)
+				}
+				if wrapper.resp == nil || wrapper.resp.Header.Get("Content-Type") != "application/vnd.kubernetes.protobuf" {
+					t.Fatalf("unexpected response: %#v", wrapper.resp)
+				}
+
+				item, err = client.Namespace(ns).Patch("test-1", types.MergePatchType, []byte(`{"metadata":{"annotations":{"foo":"baz"}}}`), metav1.PatchOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if item.Annotations["foo"] != "baz" {
+					t.Fatalf("unexpected object: %#v", item)
+				}
+
+				if err := client.Namespace(ns).Delete("test-1", &metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &item.UID}}); err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := client.Namespace(ns).Get("test-1", metav1.GetOptions{}); !apierrors.IsNotFound(err) {
+					t.Fatal(err)
+				}
+			},
+		},
+		{
+			name: "list, get, patch, and delete via metadata client on a CRD",
+			want: func(t *testing.T) {
+				ns := "default"
+
+				cr, err := crclient.Create(&unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "cr.bar.com/v1",
+						"kind":       "Foo",
+						"spec":       map[string]interface{}{"field": 1},
+						"metadata": map[string]interface{}{
+							"name": "test-1",
+							"annotations": map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					},
+				}, metav1.CreateOptions{})
```

Comment:
It would make me feel a lot better if only the creation step differed and the rest of the test was the same.

Reply:
I tightened it up a bit more, but the original intent was to allow us to add more testing here.  I didn't want to over DRY what should also be a usable example of the test client in action (i.e. ergonomics)